tsdb/store.go

```diff
@@ -613,6 +614,27 @@ func (s *Store) IteratorCreators() influxql.IteratorCreators {
 	return a
 }
 
+func (s *Store) IteratorCreator(shards []meta.ShardInfo) (influxql.IteratorCreator, error) {
+	// Generate iterators for each node.
+	ics := make([]influxql.IteratorCreator, 0)
+	if err := func() error {
+		for _, shard := range shards {
+			ic := s.ShardIteratorCreator(shard.ID)
+			if ic == nil {
+				continue
+			}
+			ics = append(ics, ic)
+		}
+
+		return nil
+	}(); err != nil {
+		influxql.IteratorCreators(ics).Close()
+		return nil, err
+	}
+
+	return influxql.IteratorCreators(ics), nil
+}
```

Comment:
Yeah, using `ShardInfo` also creates a dependency on `meta` from `tsdb` package which we've tried really hard to avoid.  `[]uint64` would be more consistent with the other functions in `Store` as well.